magicctapipe/scripts/lst1_magic/semi_automatic_scripts/list_from_h5.py

```diff
@@ -297,18 +332,27 @@ def main():
     df_MAGIC1 = df_MAGIC1[df_MAGIC1["DATE"].isin(list_date_LST_low)]
     df_MAGIC2 = df_MAGIC2[df_MAGIC2["DATE"].isin(list_date_LST_low)]
 
-    clear_files(source_in, source_out, df_LST, df_MAGIC1, df_MAGIC2)
+    clear_files(source_in, source_out, df_LST, df_MAGIC1, df_MAGIC2, allowed_M_tels)
 
-    list_run(source_in, source_out, df_LST, skip_LST, skip_MAGIC, True)
+    list_run(source_in, source_out, df_LST, skip_LST, skip_MAGIC, True, allowed_M_tels)
 
     df_MAGIC2 = magic_date(df_MAGIC2)
     df_MAGIC1 = magic_date(df_MAGIC1)
 
     M1_runs = df_MAGIC1["Run ID"].tolist()
-    if (len(M1_runs) == 0) or (len(df_MAGIC2) == 0):
+    if len(allowed_M_tels) == 2 and (len(M1_runs) == 0) or (len(df_MAGIC2) == 0):
         print("NO MAGIC stereo run found. Exiting...")
         return
-    list_run(source_in, source_out, df_MAGIC2, skip_LST, skip_MAGIC, False, M1_runs)
+    list_run(
+        source_in,
+        source_out,
+        df_MAGIC2,
```

Comment:
I would add before this line: 
df_MAGIC=df_MAGIC2 if 2 in allowed_M_tels else df_MAGIC1
and then used df_MAGIC in this line. 
This way it will also work for LST1+M1 data